module/os/linux/spl/spl-err.c

```diff
@@ -86,6 +86,9 @@ vcmn_err(int ce, const char *fmt, va_list ap)
 {
 	char msg[MAXMSGLEN];
 
+#ifndef _KERNEL
+	(void) fflush(stdout);
+#endif
```

Comment:
Since everything printed by `vcmn_err` is sent to stderr this doesn't seem like quite the right place to me.  What if we instead added the `fflush()` to `zdb` after printing the blkptrs.  Are we losing the `stdout` buffer in your case because `zdb` is  terminating unexpectedly?

Also, I believe you meant to add this to `lib/libzpool/kernel.c:vcmn_err()` which is the user space version of this function.  In this file, `_KERNEL` will always be defined.